File: 주유소.py
import sys

input = lambda: sys.stdin.readline().rstrip()

def solution(n, edges, costs):
  answer = 0 

  minCost = 1_000_000_000
  for i in range(n - 1):
    minCost = min(minCost, costs[i])
    answer += edges[i] * minCost

  return answer

N = int(input())
EDGES = list(map(int, input().split()))
COSTS = list(map(int, input().split()))
print(solution(N, EDGES, COSTS))

File: kotlin/주유소.kt
fun main(args: Array<String>) {
  val n = readln().toInt()
  val edges = readln().split(" ").map { it.toInt() }
  val costs = readln().split(" ").map { it.toInt() }

  val result = solution(n, edges, costs)
  println(result)
}

fun solution(n: Int, edges: List<Int>, costs: List<Int>): Int {
  var answer = 0

  var minCost = Int.MAX_VALUE
  for (i in 0..(n - 2)) {
    if (costs[i] < minCost) {
      minCost = costs[i]
    }

    answer += edges[i] * minCost
  }
  
  return answer
}